### ISSUE TITLE
Story: Tracking Growth: Journal

### ISSUE THREAD

**Issue by z-schwartz** [IITIDIDX597/F2021_Team2#18]
PROJECT GOAL
Community Catalyst connects a resident in the community with partners that help transform the collective vision for the neighborhood into action ensuring a more equitable and sustained community recovery.

THE REQUISITE SUB-HILL STATEMENT

Grouping: Tracking Growth

1. The volunteer can see milestones achieved by the actual organizations they have volunteered over time
2. Volunteers can see their activity and experiences formatted automatically into a CV
3. Volunteer can access private journaling tools to help them reflect and keep up with their progress

THE STORY FOR REVISION

**As a:** Volunteer

**I want:** to have a personal journal  

**So that:** I can reflect on my past volunteering initiatives

**Comment by z-schwartz** [comment]
- [x] @Tosgirl - as discussed, please revise/refine/develop the story above two more times before tomorrow at 5.

**Comment by Tosgirl** [comment]
2nd iteration: 

As a : Volunteer

I want: to keep a record of my past volunteering activities  

So that : I can measure my level of growth

**Comment by Tosgirl** [comment]
3rd Iteration

As a : Volunteer

I want: write down all my involvements as a volunteer  

So that : I can use it as a good resource for future job

**Comment by akallish** [comment]
> As a : Volunteer
> 
> I want: to keep a record of my past volunteering activities
> 
> So that : I can measure my level of growth

This may be the  final iteration. @z-schwartz maybe it can be refined to integrate reflection and future employment?

**Comment by z-schwartz** [comment]
This can include: 

Measure my growth | Update my CV | Curate Volunteer opportunities

Possibly overlaps with other issues. 

Do we split into three separate?

**Comment by UjjwalAS** [comment]
I think we can integrate update CV into both and have two as measure my growth and curate volunteer opportunities

**Comment by F2021MartinRon** [comment]
As a : Volunteer
I want: to keep a record of my past volunteering activities in a CV on the platform
So that : I can measure my level of growth

**Comment by akallish** [comment]
> I want: to keep a record of my past volunteering activities in a CV on the platform
> So that : I can measure my level of growth

@F2021MartinRon How will a CV record measure a person's level of growth. What does measure mean?

**Comment by Tosgirl** [comment]
@UjjwalAS iteration 1, @z-schwartz  iteration 2

**Comment by UjjwalAS** [comment]
SELECTED STORY

As a : Volunteer
I want: to keep a record of my past volunteering activities in a CV on the platform
So that : I can measure my level of growth



ACCEPTANCE CRITERIA ITERATION 1

Given: I’m a user logged into the Community Catalyst application
When: I tap onto "CV" button on my Home screen 
Then: I should be able to see the chronological record of all my past volunteering activities with  specific details such as 1.dates, 2.who the partner was and whether I volunteered more than once for them, 3.no. of days volunteered, 4.nature of volunteering work, 5.impactS it created on community and people  

@z-schwartz this is ready for second iteration

**Comment by akallish** [comment]
@UjjwalAS good iteration 0. @z-schwartz thanks for weighing in. What is missing to me are the details that lead to measuring growth. Are these five things the measures?

**Comment by UjjwalAS** [comment]
@akallish yes these 5 measures are aspects of growth. However I did think around the details that lead to measure growth but with that the acceptance criteria was getting split into two parts. 

1. A more concrete aspect like what I have mentioned in iteration 1. Which gets into very specific details and aspects on which growth will be measured

2. A slight abstraction where user can see growth in a graph form based on the past volunteering activities. Idea that I had was - having a stats button within CV section that shows whether the volunteer's progress is upward, downward or constant. 

I decided to go ahead with option 1 because it was capturing the details related to measuring growth without abstraction.

But stats/ graphs could be another feature that captures all the details in the background and projects only a conclusive growth pattern to users.

**Comment by akallish** [comment]
> But stats/ graphs could be another feature that captures all the details in the background and projects only a conclusive growth pattern to users.

@UjjwalAS I would integrate this as growth is based on insight or a deeper understanding.

**Comment by z-schwartz** [comment]
> SELECTED STORY
> 
> As a : Volunteer
> I want: to keep a record of my past volunteering activities in a CV on the platform
> So that : I can measure my level of growth
> 
 ACCEPTANCE CRITERIA ITERATION 2

**Given:** I’m a user logged into the Community Catalyst application
**When:** I tap onto the "CV" button on my Home screen
**Then**: I see a holistic report of my activities that shows general trends and insights into my growth in addition to specific details for each session. 

This may include: 

**1.** Dates
**2** Who the partner was
**3.** # of times I've volunteered for them
**4.** Frequency of volunteering for them
**5.** Description of contribution
**6.** Impact of contribution
**7.** The team I volunteered with
**8.** Photos

@UjjwalAS - I trimmed up the "then" section and integrated @akallish suggestion about insights + growth. I added a few "specific details" to the list and spaced a few you had out for clarity. Can you review and let me know what you think?

**Comment by akallish** [comment]
@z-schwartz @UjjwalAS great conversation. Believe the additional billeted detail that is separate from the main acceptance criteria helps

**Comment by UjjwalAS** [comment]
@z-schwartz this seems fine to me👍

**Comment by F2021MartinRon** [comment]
I like the bulleted list of details, do we move this one to done?

**Comment by UjjwalAS** [comment]
yes, I am OK. seems fine to me too

**Comment by gauribhatt** [comment]
Seems like we have done only 2 iterations on this one. Should we do one more?

**Comment by akallish** [comment]
@gauribhatt Yes

**Comment by UjjwalAS** [comment]
SELECTED STORY

As a : Volunteer
I want: to keep a record of my past volunteering activities in a CV on the platform
So that : I can measure my level of growth

ACCEPTANCE CRITERIA ITERATION 3

Given: I’m a user logged into the Community Catalyst application
When: I tap onto "CV" button or "STATISTICS" button depending on my preference
Then: I either see a record of my past volunteering activities in a CV format or a  holistic report that shows general trends and insights into my growth in addition to specific details for each session.

This may include:

1. Dates
2 Who the partner was
3. no of times I've volunteered for them
4. Frequency of volunteering for them
5. Description of contribution
6. Impact of contribution
7. The team I volunteered with
8. Photos

@akallish @z-schwartz @gauribhatt @ronjmartin @Tosgirl 
How are your views on this

In my view this third iteration captures data at two level-

1. Just a record of past volunteering activities in a CV format with the above mentioned details

and

2. A more detailed and holistic report communicating insights **(in graphical and statistical way)** such as 
- graphs showing who I have volunteered for the most and least
- causes I have never volunteered for   
- what kind of causes have I generally (repetitively) been taking in the past (in a statistical format)
- which volunteering activities created the most impact vs which one created the least

**Comment by akallish** [comment]
@UjjwalAS This is great. You really are capturing the details of the data that creates the dashboards. Remember, don't load it up with features and functions. You are testing the story, not all of CC!

@z-schwartz thoughts

**Comment by z-schwartz** [comment]
**@UjjwalAS @akallish -- I made a slight edit in the acceptance criteria, specically in the when section. I swapped in "my dashboard button. Additionally I did some slight reformatting.** 

**Given:** I’m a user logged into the Community Catalyst application
**When:** I tap onto "my dashaboard" button
**Then:** I either see a record of my past volunteering activities in a CV format OR a holistic report that shows general trends and insights into my growth in addition to specific details for each session.

Each section may contain the following. These are not directives, but ideas for consideration. 

_**CV FORMAT:** record of past volunteering activities in a CV format which may include:_

- Dates
- Who the partner was
- no of times I've volunteered for them
- Frequency of volunteering for them
- Description of contribution
- Impact of contribution
- The team I volunteered with
- Photos


_**Holistic Visualization:** more detailed and holistic report communicating insights **(in graphical and statistical way)**_ 
- graphs showing who I have volunteered for the most and least
- causes I have never volunteered for
- what kind of causes have I generally (repetitively) been taking in the past (in a statistical format)
- which volunteering activities created the most impact vs which one created the least

**Comment by akallish** [comment]
@z-schwartz you are the product manager. You took your teams improvements and integrated them.

**Comment by z-schwartz** [comment]
@akallish - Thanks, Adam. Far more succinct and clear way to say it.
